cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -233,4 +235,21 @@ private void deleteDiskEncryptionSetOnCloud(AzureClient azureClient, String desR
             }
         });
     }
+
+    private void removeKeyVaultAccessPolicyFromDiskEncryptionSetServicePrincipal(AzureClient azureClient, String desResourceGroupName, String desName,
+            String encryptionKeyUrl, String desPrincipalObjectId) {
+        String vaultName;
+        Matcher matcher = ENCRYPTION_KEY_URL_VAULT_NAME.matcher(encryptionKeyUrl);
+        if (matcher.matches()) {
+            vaultName = matcher.group(1);
+        } else {
+            throw new IllegalArgumentException(String.format("Failed to deduce vault name from given encryption key url \"%s\"", encryptionKeyUrl));
+        }
+        String description = String.format("access to Key Vault \"%s\" in Resource Group \"%s\" for Service Principal having object ID \"%s\" " +
+                        "associated with Disk Encryption Set \"%s\" in Resource Group \"%s\"", vaultName, desResourceGroupName, desPrincipalObjectId,
+                desName, desResourceGroupName);
+                LOGGER.info("Removing {}.", description);
+                azureClient.removeKeyVaultAccessPolicyFromServicePrincipal(desResourceGroupName, vaultName, desPrincipalObjectId);
+                LOGGER.info("Removed {}.", description);
```

Comment:
I know it is overly paranoid, but I would perhaps wrap the these lines in a `retryService.testWith2SecDelayMax15Times()` call, and inside the lambda add a `try` / `catch` around like for `grantKeyVaultAccessPolicyToDiskEncryptionSetServicePrincipal()` and `deleteDiskEncryptionSetOnCloud()`. The vault name extraction (lines 241-247) can stay outside of the retry logic, as there is no point retrying if the key URL is malformed (which should not happen in practice at this point).